src/util/merkleblock.rs

```diff
@@ -0,0 +1,773 @@
+// Rust Bitcoin Library
+// Written by
+//   John L. Jegutanis
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+//
+// This code was translated from merkleblock.h, merkleblock.cpp and pmt_tests.cpp
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+//! Merkle Block and Partial Merkle Tree
+//!
+//! Support proofs that transaction(s) belong to a block.
+//!
+//! # Examples
+//!
+//! ```ignore
```

Comment:
About all the code samples: I usually start with 
```
` ` `rust
```
(ignore the spaces)

This makes for higher likelyhood of correct coloring with different tools. I would expect all the Rust docs-focused tools to infer Rust syntax, but you never know.

Reply:
You are right about the `rust` attribute but I had issues passing this doc test for Rust < 1.32.0 (for newer versions it works just fine).

I think it is because there is a `extern crate bitcoin_hashes;` where older versions could not handle well.

With the `ignore` attribute, the test does not run at all and that sucks but at least it passes the doc test on the ancient Rust 1.22.0 that we still support.